### ISSUE TITLE
v3.0.5 not compatible with macOS Monterey (12.x)

### ISSUE THREAD

**Issue by DanBurkhardt** [KinesisCorporation/SmartSetApps#7]
Hello,

The downloadable version [on the website](https://gaming.kinesis-ergo.com/fs-edge-rgb-support/) appears not to be compatible with the permissions flow for macOS Monterey. 
I am running version 12.2.1 (latest as of today).
I originally tried to use the old 2.x.x version and it threw a warning, so I updated to v3.0.5

Device: Freestyle Edge RGB KB975-BRN

Steps to troubleshoot:
 
- Update to the latest keyboard and lighting firmware
- Mount v-drive using smartset+F8
- The drive mounts perfectly fine
- Open application
- Application cannot detect the drive
- Repeated attempts to scan for drive failed

Thoughts:

- If the software wanted to read a removable disk, it would normally have to ask permission in the latest versions of macOS
- I think the application doesn't have permissions necessary to read removable drives
- I cannot grant access to the application manually in the (files and folders) section of macOS permissions settings
- No permissions request screen pops up instructing me to grant the application permission

**Comment by KinesisCorporation** [comment]
Hi Dan, Thanks for bringing this to our attention. We just re-tested with Monterey and when we launched the App we got a prompt to allow the SmartSet App to access the removable drive. Please shoot us a ticket and so we can get more info and can work with you directly.

https://gaming.kinesis-ergo.com/contact-tech-support/

**Comment by DanBurkhardt** [comment]
Ok will do. I am never shown this permission but will send a  support ticket.

**Comment by DanBurkhardt** [comment]
I have one more detail @KinesisCorporation: 

I am running a 2019 Mac Pro and my home folder / all data is actually technically stored on a "removable drive". Let me explain:

- For some ungodly reason, Apple has decided that drives attached through the PCIe expansion slot should be treated at the OS level like removable drives

- My home folder and all data is on a PICe m.2 RAID 0, as seen here: https://www.sonnettech.com/product/m2-4x4-pcie-card/techspecs.html

I think that the application may not be taking this particular weird scenario into consideration:
- Application launched from a home folder applications drive that itself is on a removable disk
- Somehow cannot ask for removable disk access because it already is on one

That's my best guess. I have submitted a ticket and referenced it here. I can't update it, so can you please pass this along to support?

Thanks, 
Dan

**Comment by rodneytamblyn** [comment]
Hi Kinesis team,

Also reporting here that I am unable to get SmartSet to detect v-Drive on Monterray.  Mac Mini Monterray 12.2.  V-drive mounts when Program+F8 clicked, but detect drive fails to detect.  Can you confirm if you are able to reproduce this?

~ Rod

**Comment by KinesisCorporation** [comment]
Hi Rod,

Sorry to hear you are having problems. Can you submit a trouble ticket and we can help you from there? We don't have the resources to troubleshoot through GitHub. Thanks!

https://gaming.kinesis-ergo.com/contact-tech-support/

**Comment by ericdrobinson** [comment]
I appear to have the same issue with v3.0.5 on macOS 13.0 (Ventura) on a 2021 MacBook Pro M1 Max. 

The v-Drive identifies as "FS PRO" when mounted. The keyboard itself is connected to the computer through a CalDigit [Element Hub](https://www.caldigit.com/thunderbolt-4-element-hub/). No idea if this factors in at all.

I have already submitted a ticket to support.

**Comment by ericdrobinson** [comment]
> The v-Drive identifies as "FS PRO" when mounted.

This was the problem. The SmartSet App identifies devices by **the name of the mounted v-Drive**. The version of the application for the Freestyle Edge RGB [expects](https://github.com/KinesisCorporation/SmartSetApps/blob/fd4a17217aebbfc8a793fab2c70fc8eb687bdd18/SmartSetMaster/u_form_dashboard.pas#L333) the name to be "[FS EDGE RGB](https://github.com/KinesisCorporation/SmartSetApps/blob/fd4a17217aebbfc8a793fab2c70fc8eb687bdd18/Common/u_const.pas#L771)". In my case, the "FS PRO" name appears to be related to the [Freestyle Pro](https://kinesis-ergo.com/shop/freestyle-pro/) keyboard.

In my case, I purchased the keyboard refurbished from Kinesis. I'd guess that the name mixup happened sometime during the refurbishing process.

Either way, simply renaming the v-Drive to "FS EDGE RGB" in the Finder enabled the SmartSet app to identify the keyboard and work as expected. 👍

**Comment by theAkito** [comment]
> This was the problem. The SmartSet App identifies devices by **the name of the mounted v-Drive**. The version of the application for the Freestyle Edge RGB [expects](https://github.com/KinesisCorporation/SmartSetApps/blob/fd4a17217aebbfc8a793fab2c70fc8eb687bdd18/SmartSetMaster/u_form_dashboard.pas#L333) the name to be "[FS EDGE RGB](https://github.com/KinesisCorporation/SmartSetApps/blob/fd4a17217aebbfc8a793fab2c70fc8eb687bdd18/Common/u_const.pas#L771)". In my case, the "FS PRO" name appears to be related to the [Freestyle Pro](https://kinesis-ergo.com/shop/freestyle-pro/) keyboard.

Sorry about that. I wanted to open an issue about this like a year ago.

Not even the Kinesis support knew about this. About a year ago, it took me a couple of hours to accidentally stumble upon this.

**Comment by dylan-jh** [comment]
Correct, the SmartSet App identifies the keyboard based on the name of the removal drive when its mounted to your PC. There are 3 different Freestyle devices supported by slightly different versions of the App.

- KB950 (Original Freestyle Edge): "FS EDGE"> https://gaming.kinesis-ergo.com/fs-edge-support/#smartset
- KB900 (Freestyle Pro): "FS PRO"> https://kinesis-ergo.com/support/freestyle-pro/#smartset-app
- KB975 (Freestyle Edge RGB): "FS EDGE RGB"> https://gaming.kinesis-ergo.com/fs-edge-rgb-support/#smartset

**Comment by ericdrobinson** [comment]
That's great information, thanks for adding it!

The most important takeaway for users finding their way here when the SmartApp fails to detect their keyboard is that they need to **first** verify that the name of the v-Drive that gets mounted matches the expected name for their keyboard model.

As suggested [here](https://github.com/KinesisCorporation/SmartSetApps/issues/9), this information should probably be added to the FAQ/Troubleshooting/Manual resources for each keyboard.
